lib/manager/nuget/package-tree.ts

```diff
@@ -0,0 +1,99 @@
+import path from 'path';
+import util from 'util';
+import { glob } from 'glob';
+import Graph from 'graph-data-structure';
+import upath from 'upath';
+import { DOMParser as dom } from 'xmldom';
+import xpath from 'xpath';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import { readLocalFile } from '../../util/fs';
+import { regEx } from '../../util/regex';
+
+const globAsync = util.promisify(glob);
+
+// Get all package files at any level of ancestry that depend on `packageFileName`
+export async function getDependentPackageFiles(
+  packageFileName: string
+): Promise<string[]> {
+  const { localDir } = GlobalConfig.get();
+  if (localDir === undefined) {
+    throw new Error('localDir must be set');
+  }
+  const packageFiles = await getAllPackageFiles(localDir);
+  const graph: any = Graph();
```

Comment:
We need types for this!

Reply:
I'm no Typescript expert. Does `ReturnType<typeof Graph>` work here?

The types are defined here: https://github.com/datavis-tech/graph-data-structure/blob/master/index.d.ts